blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScm.java

```diff
@@ -343,11 +343,23 @@ private static StandardUsernamePasswordCredentials findUsernamePasswordCredentia
                         Jenkins.getAuthentication(),
                         URIRequirementBuilder.fromUri(scm.getUri()).build()),
                 CredentialsMatchers.allOf(CredentialsMatchers.withId(scm.getId()),
-                        CredentialsMatchers.anyOf(CredentialsMatchers.instanceOf(StandardUsernamePasswordCredentials.class)))
+                        CredentialsMatchers.withScope(CredentialsScope.USER))
         );
     }
 
-    private static @CheckForNull StandardUsernamePasswordCredentials findUsernamePasswordCredential(String id){
+    static StandardUsernamePasswordCredentials findUsernamePasswordCredential(String uri, String credentialId){
+        return CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        StandardUsernamePasswordCredentials.class,
+                        Jenkins.getInstance(),
+                        Jenkins.getAuthentication(),
+                        URIRequirementBuilder.fromUri(uri).build()),
+                CredentialsMatchers.allOf(CredentialsMatchers.withId(credentialId),
+                        CredentialsMatchers.withScope(CredentialsScope.USER))
```

Comment:
Why is that? I want to keep it strict and if there is GitHub access key based credentials it better come from USER scope or we reject it.